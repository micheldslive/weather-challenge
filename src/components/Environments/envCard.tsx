type EnvCardProps = {
  title?: string
  text?: string
}

export const EnvCard = ({ title, text }: EnvCardProps) => {
  return (
    <div className='envCard'>
      <p aria-label='title' className='title'>
        {title}
      </p>
      <span aria-label='subtitle' className='subtitle'>
        {text}
      </span>
    </div>
  )
}
