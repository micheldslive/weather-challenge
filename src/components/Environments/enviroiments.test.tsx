import { getRenderer } from '@/core/tests/helpers'
import { EnvCard } from './envCard'

describe('<EnvCard />', () => {
  it('should render the title and subtitle correctly', () => {
    const { getByLabelText, container } = useRenderer

    expect(getByLabelText('title')).toBeInTheDocument()
    expect(getByLabelText('subtitle')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})

const useRenderer = getRenderer(<EnvCard />)
