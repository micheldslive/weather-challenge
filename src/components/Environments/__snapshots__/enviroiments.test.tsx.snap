// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnvCard /> should render the title and subtitle correctly 1`] = `
<div
  class="envCard"
>
  <p
    aria-label="title"
    class="title"
  />
  <span
    aria-label="subtitle"
    class="subtitle"
  />
</div>
`;
